tutorials/google-kubernetes-engine-hyperparameter-search/index.md

```diff
@@ -0,0 +1,266 @@
+
+Follow the steps in the [Before you start section][beforeyoustart] to install dependencies.
+
+## Overview
+
+Hyperparameter search is one of the time consuming part of fitting a machine learning model.  A model is fitted to the data many times, each time with different hyperparameters and the performance recorded.  This can be parallelized in the following ways:
+
+- Many scikit-learn models have the `n_jobs` argument allowing multi-thread model fitting, for instance [RandomForestClassifier][rfc].  
+
+- The wrappers such as [RandomizedSearchCV][randomizedsearchcv] that handle hyperparameter search also have their `n_jobs` argument running separate model fitting jobs on multiple subprocesses in parallel.
+
+Running fitting jobs in parallel speeds up the search, but could require a lot of computational resources.  The [sample notebook][gke_randomized_search] provides a workflow that allows you to set up hyperparameter search experiments on your computer but sends the actual workload to a cluster on Google Kubernetes Engine.  The advantage of doing this include:
+
+- You can continue to use your laptop/workstation for other work while waiting for the results.
+
+- You can use more powerful machines to speed up the search, for instance mulitple nodes with 64 virtual CPU cores.
+
+To accomplish this, we will create a `SearchCV` object in the notebook, upload a pickled copy of this object to Cloud Storage.  A job running on a cluster which we will create then retrieves that pickled object and calls its `fit` method and saves the fitted object to Cloud Storage.  Finally, we can download the fitted object and call its prediction method or examine evaluation metrics.
+
+## A closer look
+
+### Overview of the workflow
+
+The sample is designed to keep most of the workflow in a Jupyter notebook and to feel very similar to a typical experimentation workflow.  To run a scikit-learn hyperparameter search job locally on your laptop you might run something like this (snippets from [sample notebook][gke_randomized_search]):
+
+```python
+rfc = RandomForestClassifier(n_jobs=-1)
+param_distributions = {
+    'max_features': uniform(0.6, 0.4),
+    'n_estimators': range(20, 401, 20),
+}
+search = RandomizedSearchCV(estimator=rfc, param_distributions=param_distributions, n_iter=100, n_jobs=-1)
+```
+
+And follow it with a `search.fit(X, y)` call.  
+
+With the helper modules included in this sample, you can send the computation workload to Kubernetes engine by wrapping the `SearchCV` object and call its `fit` method:
+
+```python
+from gke_parallel import GKEParallel
+
+gke_search = GKEParallel(search, project_id, zone, cluster_id, bucket_name, image_name)
+gke_search.fit(X, y)
+```
+
+(There are additional steps of creating a docker image and a cluster that we need to run before calling `gke_search.fit`.)
```

Comment:
Suggested change for clarification:
(For the purposes of this overview, we skipped a few additional steps for creating a docker image and a cluster that we need to run before calling `gke_search.fit`.)